web-registry/src/components/atoms/WithLoader.tsx

```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { CircularProgress } from '@mui/material';
+
+type Props = {
+  isLoading: boolean;
+  children: JSX.Element;
+};
+
+const WithLoader = ({ isLoading, children }: Props): JSX.Element => {
+  if (isLoading) {
+    return <CircularProgress color="secondary" />;
```

Comment:
We have a `<Loading />` component already, but I like the utility in this more. Seems like `WithLoader` could replace the current areas we're using `Loading`... probably don't need both.

Also wonder if we should wrap the progress in a `Box` we can pass SX to, to help with layout thrashing if `children` have an assigned height etc (similar to the current `Loading` component)